src/Lecture3.hs

```diff
@@ -134,6 +151,7 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
+  (<>) (List1 x xs) (List1 y ys)  = List1 x (xs <> [y] <> ys)
```

Comment:
You can slightly improve the performance here by using the `:` operator instead of list append `<>` or `++` 🐎 
```suggestion
  (<>) (List1 x xs) (List1 y ys) = List1 x (xs <> (y : ys))
```